d2/src/test/java/com/linkedin/d2/xds/TestXdsClientImpl.java

```diff
@@ -982,8 +1002,10 @@ private static final class XdsClientImplFixture
         setResourceVersions(useGlobCollections);
       }
 
+      _executorService = spy(Executors.newScheduledThreadPool(1));
```

Comment:
FYI for reviewers: I changed the executor from `Executors.newSingleThreadScheduledExecutor(new NamedThreadFactory("test executor")` to `Executors.newScheduledThreadPool(1)` because `spy` didn't work on the former executor

```
Mockito cannot mock this class: class java.util.concurrent.Executors$DelegatedScheduledExecutorService
Mockito can only mock visible & non-final classes.
```

As far as I'm aware, these should be equivalent